src/jmcomic/jm_task_context.py

```
@@ -0,0 +1,59 @@
+    context.update(fields)
+    token = JM_TASK_CONTEXT.set(MappingProxyType(context))
+    try:
+        yield
+    finally:
+        JM_TASK_CONTEXT.reset(token)
+
+
+def bind_jm_task_context(func: Callable, context: Optional[Mapping] = None) -> Callable:
+    """Bind a synchronous callable to a snapshot of the current task context."""
+    if (inspect.iscoroutinefunction(func)
+            or inspect.iscoroutinefunction(getattr(func, '__call__', None))):
```

Ruff (0.16.0)

[warning] Using `hasattr(x, "__call__")` to test if x is callable is unreliable. Use `callable(x)` for consistent results.

(B004)